[Bug]: liboqs build fails with clang

### Contact Details

_No response_

### Version

current master (commit 9db20774d8abe0725734d48090bd4f650a477b8a)

### Description

Building wolfssl with liboqs fails with clang (I'm using Clang 15) due to an implicit casting in `wolfcrypt/src/port/liboqs/liboqs.c`. Possibly the consequence of https://github.com/wolfSSL/wolfssl/pull/7026

### Reproduction steps

export CC=clang
git clone https://github.com/open-quantum-safe/liboqs.git
git clone --depth 1 https://github.com/wolfSSL/wolfssl.git
mkdir liboqs-install/
cd liboqs/
git checkout 0.8.0
mkdir build/
cd build/
cmake -DCMAKE_INSTALL_PREFIX:PATH=$(realpath ../../liboqs-install) -DOQS_USE_OPENSSL=0 ..
make -j$(nproc)
make install
cd ../../wolfssl/
autoreconf -ivf
./configure --enable-static --disable-examples --disable-crypttests --enable-aesccm --enable-aescfb --enable-aesctr --enable-aeseax --enable-aesgcm-stream --enable-aeskeywrap --enable-aesofb --enable-aessiv --enable-arc4 --enable-asn=original --enable-blake2 --enable-blake2s --enable-camellia --enable-certext --enable-cmac --enable-compkey --enable-crl --enable-cryptocb --enable-curve25519 --enable-curve448 --enable-des3 --enable-dsa --enable-dtls --enable-dtls13 --enable-dtlscid --enable-earlydata --enable-ecccustcurves --enable-ecccustcurves=all --enable-eccencrypt --enable-eccsi --enable-ed25519 --enable-ed25519-stream --enable-ed448 --enable-ed448-stream --enable-harden --enable-hkdf --enable-hrrcookie --enable-indef --enable-keygen --enable-md2 --enable-md4 --enable-nullcipher --enable-ocsp --enable-ocspstapling --enable-oldtls --enable-postauth --enable-psk --enable-pwdbased --enable-ripemd --enable-scrypt --enable-secure-renegotiation --enable-session-ticket --enable-shake128 --enable-shake256 --enable-siphash --enable-smallstack --enable-sni --enable-srp --enable-srtp --enable-sslv3 --enable-tls13 --enable-tlsv10 --enable-tlsx --enable-x963kdf --enable-xchacha --enable-xts --with-eccminsz=0 --enable-hpke --enable-ocspstapling2 --enable-pkcs7 --enable-dtls-frag-ch --with-liboqs=$(realpath ../liboqs-install/)
make

### Relevant log output

```shell
wolfcrypt/src/port/liboqs/liboqs.c:53:62: error: implicit conversion loses integer precision: 'size_t' (aka 'unsigned long') to 'word32' (aka 'unsigned int') [-Werror,-Wshorten-64-to-32]
    int ret = wc_RNG_GenerateBlock(liboqsCurrentRNG, buffer, numOfBytes);
              ~~~~~~~~~~~~~~~~~~~~                           ^~~~~~~~~~
1 error generated.
```

--- a/wolfcrypt/src/port/liboqs/liboqs.c
+++ b/wolfcrypt/src/port/liboqs/liboqs.c
@@ -50,7 +50,7 @@ static int liboqs_init = 0;
 
 static void wolfSSL_liboqsGetRandomData(uint8_t* buffer, size_t numOfBytes)
 {
-    int ret = wc_RNG_GenerateBlock(liboqsCurrentRNG, buffer, numOfBytes);
+    int ret = wc_RNG_GenerateBlock(liboqsCurrentRNG, buffer, (word32)numOfBytes);
     if (ret != 0) {
         // ToDo: liboqs exits programm if RNG fails, not sure what to do here
     }

